pkg/rpcservice/middleware.go

```diff
@@ -14,6 +16,23 @@ import (
 
 const requiredHeader = "sec-x-stripe-cli"
 
+// WrappedServerStream wraps a ServerSteam so that we can pass values through context.
+// https://pkg.go.dev/github.com/grpc-ecosystem/go-grpc-middleware#hdr-Writing_Your_Own
+type WrappedServerStream struct {
+	grpc.ServerStream
+	ctx context.Context
+}
+
+// Context returns the context for this stream.
+func (w WrappedServerStream) Context() context.Context {
+	return w.ctx
```

Comment:
does it break `WithTelemetryClient` if `w.ctx` is `null`? or can it be `null`?

Reply:
If `w.ctx` is nil, this function would return a `nil` to the caller to handle.

Reply:
On second thought, I don't see a problem with checking here and passing in a `context.Background()` if for some reason the context ends up being nil. I'll make that update